src/main/kotlin/io/cloudflight/gradle/autoconfigure/swagger/SwaggerApiConfigurePlugin.kt

```
@@ -0,0 +1,180 @@
+                    it
+                )
+            }
+
+            with(documentationTask) {
+                group = AutoConfigureGradlePlugin.TASK_GROUP
+                classFinder = ClassFinder(target)
+                readerFactory = ReaderFactory(classFinder, ClassFinder(target, javaClass.classLoader))
+                generatorFactory = GeneratorFactory(classFinder)
+                apiSourceValidator = ApiSourceValidator(
+                    InfoValidator(LicenseValidator()), SecurityDefinitionValidator(
+                        ScopeValidator()
+                    ), TagValidator(ExternalDocsValidator())
+                )
+
+                inputFiles = getFilesFromSourceSet(target)
+                outputDirectories = listOf(target.file(swagger.apiSourceExtensions.first().swaggerDirectory))
+                outputFile = outputs.map { it.file }
+
+                doFirst(ConfigureSwaggerAction)
+            }
+
+            jarTask.dependsOn(documentationTask)
+        }
+    }
+
+    internal object ConfigureSwaggerAction : Action<Task> {
+        override fun execute(t: Task) {
+            val project = t.project
+            val swagger = project.extensions.getByType(SwaggerExtension::class.java)
+            val extension = swagger.apiSourceExtensions.first()
+            with(extension) {
+                info.version = project.version.toString()
+                if (locations == null) {
+                    locations = listOf(
+                        project.group.toString() + ".api",
+                        project.group.toString() + ".api.dto",
+                    )
+                }
+            }
+
+            // Workaround for an issue in the generateSwaggerDocumentation plugin
+            // check https://github.com/gigaSproule/swagger-gradle-plugin/issues/158#issuecomment-585823379
+            val classLoader = getUrlClassLoader(project.buildscript.classLoader)
+            if (classLoader != null) {
```

We should throw an Exception if there is no classloader found